### PR TITLE
feat(broker-redis): add support to nestjs v10 

### PR DESCRIPTION
**Pull request by overbit**
Fixes: amplication/amplication#7239

## PR Checklist

- [x] Tests for the changes have been added
- [x] `npm build` doesn't throw any error
- [x] `npm test` doesn't throw any error

IMPORTANT: Please review the [CONTRIBUTING.md](https://github.com/amplication/amplication/blob/master/CODE_OF_CONDUCT.md) file for detailed contributing guidelines.

### CODE REVIEW

File: plugins/broker-redis/src/constants.ts [hunk 1]
```diff
@@ -10,9 +10,8 @@ export const templatesPath = join(__dirname, "templates");
 
 export const dependencies = {
   dependencies: {
-    redis: "3.1.2",
-    "@nestjs/microservices": "8.2.3",
-    rxjs: "^7.8.1",
```
Comment (abrl91) [inline plugins/broker-redis/src/constants.ts:15]: what about `rxjs` ?

File: plugins/broker-redis/src/static/generateRedisClientOptions.ts [hunk 2]
```diff
@@ -38,13 +37,19 @@ export const generateRedisClientOptions = (
...
+      port: redisPort,
       retryAttempts: redisRetryAttempts,
       retryDelay: redisRetryDelay,
-      tls: redisEnableTLS,
     },
   };
+
+  if (redisEnableTLS) {
+    redisOptions.options!.tls = {};
+  }
```
Comment (abrl91) [inline plugins/broker-redis/src/static/generateRedisClientOptions.ts:52]: Can we really be sure that we do have `redisOptions.options`
why not to do something like that
```suggestion
  if (redisEnableTLS && redisOptions.options) {
    redisOptions.options.tls = {};
  }
```

File: plugins/broker-redis/src/constants.ts (hunk 1 above)
Comment (overbit) [reply plugins/broker-redis/src/constants.ts:15]: it wasn't used at all..

File: plugins/broker-redis/src/static/generateRedisClientOptions.ts (hunk 2 above)
Comment (overbit) [reply plugins/broker-redis/src/static/generateRedisClientOptions.ts:52]: good one! Committing

**Review by abrl91 — APPROVED** [review]

**Pull request merged**